packages/flutter/test/cupertino/switch_test.dart

```diff
@@ -1092,6 +1141,36 @@ void main() {
     );
   });
 
+  testWidgets('Switch configures mouse cursor', (WidgetTester tester) async {
+    const bool value = false;
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            return Center(
+              child: CupertinoSwitch(
+                value: value,
+                dragStartBehavior: DragStartBehavior.down,
+                mouseCursor: WidgetStateProperty.all(SystemMouseCursors.forbidden),
+                onChanged: (bool newValue) { },
+              ),
+            );
+          },
+        ),
+      ),
+    );
+    final TestGesture gesture = await tester.createGesture(
+      kind: PointerDeviceKind.mouse,
+      pointer: 1
+    );
+    await gesture.addPointer(location: tester.getCenter(find.byType(CupertinoSwitch)));
+    await tester.pump();
+    await gesture.moveTo(tester.getCenter(find.byType(CupertinoSwitch)));
```

Comment:
Are you moving to the same location that the pointer is already at? That's probably why you don't need a `pump` after this line currently.

Also, you should add an `expect` for the case where the cursor is not `forbidden` and not pointing at the Switch.